solr/core/src/java/org/apache/solr/handler/admin/SplitOp.java

```diff
@@ -263,8 +263,9 @@ private void handleGetRanges(CoreAdminHandler.CallInfo it, String coreName) thro
         DocCollection collection = clusterState.getCollection(collectionName);
         String sliceName = parentCore.getCoreDescriptor().getCloudDescriptor().getShardId();
         Slice slice = collection.getSlice(sliceName);
-        DocRouter router =
-            collection.getRouter() != null ? collection.getRouter() : DocRouter.DEFAULT;
+        CompositeIdRouter router =
```

Comment:
is it just a formatting change? why is this required ?

Reply:
Changed declared type to be specifically CompositeIdRouter because in fact splits depend on CompositeIdRouter in particular.  I added a method or two there as well that SplitOp calls.

Reply:
As I understand, the router was 'expected' to be a CompositeIdRouter, even if not clear here, because of the code below manipulating the terms and expecting to find a CompositeIdRouter.SEPARATOR.
It becomes clearer.
Should we add a check like checkRouterSupportsSplitKey() with a clearer exception?

Reply:
Correcting myself.  This method, `handleGetRanges`, is only called for splitByPrefix (as its javadocs say), which is what depends on CompositeIdRouter.

I suppose if there was some weird/unexpected code path (maybe in the future), a ClassCastException wouldn't be particularly unfriendly?